internal/controller/eventbus/eventlog/eventlog.go

```diff
@@ -286,10 +279,16 @@ func (mgr *eventlogManager) GetAppendableSegment(ctx context.Context,
 		s = el.currentAppendableSegment()
 	}
 
-	for len(result) < num && s != nil {
-		result = append(result, s)
-		s = el.nextOf(s)
+	list := el.listOfRight(s, true)
```

Comment:
why not returns `[]Segment` from `listOfRight`?

Reply:
fixed